_posts/2022-07-01-argocd-image-updater-for-functions.md

```diff
@@ -0,0 +1,248 @@
+The applications Chart has templates for two Argo CD applications. The openfaas-operator-app for the deployment of OpenFaaS and the openfaas-functions-app for the deployment of our functions.
+
+The functions Chart is also pictured here, it is used by the openfaas-functions app. As mentioned in the prerequisites, you can checkout our [tutorial on how to package functions with helm](https://www.openfaas.com/blog/howto-package-functions-with-helm/) to create this Chart.
+
+The template for the openfaas-operator application:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: openfaas-operator
+  namespace: argocd
+spec:
+  destination:
+    namespace: openfaas
+    server: {{ .Values.spec.destination.server }}
+  project: default
+  source:
+    helm:
+      parameters:
+      - name: operator.create
+        value: "true"
+      - name: generateBasicAuth
+        value: "true"
+      - name: functionNamespace
+        value: openfaas-fn
+    path: chart/openfaas
+    repoURL: https://github.com/openfaas/faas-netes.git
+  syncPolicy:
+    automated:
+      selfHeal: true
+```
+
+The template for the openfaas-functions application:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: openfaas-functions
+  namespace: argocd
+spec:
+  destination:
+    server: {{ .Values.spec.destination.server }}
+  project: default
+  source:
+    path: chart/functions
+    repoURL: https://github.com/welteki/openfaas-argocd-example.git
+  syncPolicy:
+    automated:
+      selfHeal: true
+```
+
+The full example is [available on GitHub](https://github.com/welteki/openfaas-argocd-example)
+
+Once 
+The parent app can be created and synced via the CLI:
+```bash
+argocd app create applications \
+    --dest-namespace argocd \
+    --dest-server https://kubernetes.default.svc \
+    --repo https://github.com/welteki/openfaas-argocd-example.git \
+    --path chart/applications
+
+argocd app sync applications
+```
+
+You can check the Argo CD UI to see if the apps were synced successfully or use `kubectl` to verify if OpenFaaS and the functions are running.
+
+```bash
+kubectl get pods -n openfaas
+
+NAME                              READY   STATUS    RESTARTS          AGE
+alertmanager-c4df79ff7-2fxcs      1/1     Running   2 (4h42m ago)     39h
+basic-auth-plugin-588f588-rkgmz   1/1     Running   2 (4h42m ago)     39h
+gateway-55fd54cb76-xf4n7          2/2     Running   3 (4h42m ago)     39h
+nats-67d8f684f8-x46zj             1/1     Running   2 (4h42m ago)     39h
+prometheus-cd4844fc7-c6j2b        1/1     Running   2 (4h42m ago)     39h
+queue-worker-5795ff9bb5-tkwpv     1/1     Running   3 (4h42m ago)     39h
+```
+
+```bash
+kubectl get functions -n openfaas-fn
+
+NAME             IMAGE
+email-notify     welteki/email-notify:0.1.0
+marketing-list   welteki/marketing-list:0.1.0
+```
+
+### Install and configure the ArgoCD Image Updater
+We install the updater as Kubernetes workload in the Argo CD namespace.
```

Comment:
as _a_ Kubernetes workload?

Reply:
This was reworded when I updated this section.